src/main/kotlin/org/team5419/fault/BerkeliumTimedRobot.kt

```diff
@@ -0,0 +1,62 @@
+package org.team5419.fault
+
+import edu.wpi.first.hal.FRCNetComm
+import edu.wpi.first.hal.HAL
+import edu.wpi.first.hal.NotifierJNI
+import edu.wpi.first.wpilibj.RobotController
+import org.team5419.fault.math.units.SIUnit
+import org.team5419.fault.math.units.Second
+import org.team5419.fault.math.units.milliseconds
+import org.team5419.fault.math.units.seconds
+
+open class BerkeliumTimedRobot(
+    period: SIUnit<Second> = kDefaultPeriod,
+    overrunDumpPeriod: SIUnit<Second> = kDefaultOverrunDumpPeriod
+) : BerkeliumIterativeRobotBase(period, overrunDumpPeriod) {
+
+    private val mNotifierHandle = NotifierJNI.initializeNotifier()
+    private var mExpirationTime = 0.0
+
+    init {
+        NotifierJNI.setNotifierName(mNotifierHandle, "TimedRobot")
+        HAL.report(FRCNetComm.tResourceType.kResourceType_Framework, FRCNetComm.tInstances.kFramework_Timed)
+        HAL.report(FRCNetComm.tResourceType.kResourceType_Language, kLanguageKotlin)
+    }
+
+    override fun startCompetition() {
+        robotInit()
+        HAL.observeUserProgramStarting()
+        mExpirationTime = RobotController.getFPGATime() * 1e-6 + mPeriod.value
+        while (true) {
+            var currentTime = NotifierJNI.waitForNotifierAlarm(mNotifierHandle)
+            if (currentTime == 0.toLong()) {
+                // this bad
+                break
+            }
```

Comment:
what's happening here?

Reply:
This is a condition that is checked in wpilib's implementation, not sure when it would actually happen, but I figured that I should put it in. Should I add an error message or something?

Reply:
Although I'm not sure if it's actually an error.

Reply:
Can you store 0 as a long, instead of making the conversion every iteration?

Reply:
@nicholasunderwood it *shouldn't* matter, as the kotlin compiler should optimize the Long class to a primitive type, which is very cheap to cast performance wise.